lib/profiler/get_tick.h

```diff
@@ -67,6 +67,19 @@ get_tick(void)
     return regs;
 }
 
+#elif defined(__APPLE__)
+#include <mach/mach_time.h>
+
+static __inline hrtime_t
+get_tick(void)
+{
+    static mach_timebase_info_data_t base;
+    uint64_t ticks;
+    (void) mach_timebase_info (&base);
```

Comment:
Static `base` suggests that we want to initialize `base` once and reuse it many times (e.g., because initialization is expensive). Unconditional mach_timebase_info() call suggests the opposite initialize-everytime approach (e.g., because initialization is expected to be done at compile time). Please pick the best single strategy and run with it.